### PR TITLE
[FrameworkBundle][Translator] scan directories for translations sequentially

### LINKED ISSUE #40014: Translation domain overrides no longer being used
**Symfony version(s) affected**: 4.4.19

**Description**  
After upgrading from 4.4.18 to 4.4.19, my custom translation messages for the `security` domain is no longer being used.

**How to reproduce**  
1. Create `translations/security.en.yml` with the following contents:
    ```yaml
    "Invalid credentials.": Custom Message!
    ```
2. My translator config looks like the following:
    ```yaml
    framework:
        default_locale: en
        translator:
            default_path: '%kernel.project_dir%/translations'
            fallbacks:
                - en
    ```
3. In a twig template:
    ```twig
    {{ 'Invalid credentials.'|trans([], 'security') }} {# resolves to "Invalid credentials." but expected "Custom Message!" #}
    ```

**Possible Solution**  
I looked though the changes from 4.4.18 to 4.4.19 but couldn't pinpoint exactly what caused this bug.

**Additional context**  
~I can work on a reproducer.~ Reproducer: https://github.com/kbond/symfony-reproducer/tree/bug/trans-domain:
- [Commit demonstrating issue on 4.4.19](https://github.com/kbond/symfony-reproducer/commit/c87fd1b94780b92b98e097dae16ba99640acc2ea)
- [Commit that fixes issue (by downgrading to 4.4.18)](https://github.com/kbond/symfony-reproducer/commit/0c4364822b43b62c333b02c01e84b6fcb30bf1f7)

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40014
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by jderusse — DISMISSED** [review]

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -1236,24 +1236,26 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
...
+                    ->followLinks()
+                    ->files()
+                    ->filter(function (\SplFileInfo $file) {
+                        return 2 <= substr_count($file->getBasename(), '.') && preg_match('/\.\w+$/', $file->getBasename());
+                    })
+                    ->in($dir)
+                    ->sortByName()
+                ;
+                foreach ($finder as $file) {
+                    $fileNameParts = explode('.', basename($file));
```
Comment (rheh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:1251]: Is there a reason for using basename over $file->getBasename()?

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -1236,24 +1236,26 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
...
+                    $locale = $fileNameParts[\count($fileNameParts) - 2];
```
Comment (rheh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:1252]: Suggestion : Wouldn't $info->getBasename($info->getExtension())) give us the locale? Assuming the filer is only matching patterns such as 'ar.json'?  I would remove the need for the explode and fileparts?

**Comment by xabbuh** [comment]
@jderusse test added

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 2 above)
Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:1252]: That code was here before. I suggest not to change it as part of this bugfix.

**Review by jderusse — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**